Update mistral-medium prices and context sizes

regismesquita opened the pull request:
While testing the Mistral model, I noticed a discrepancy in the pricing shown on the logs screen. After reviewing the code, I confirmed that the pricing values were incorrect.

This PR corrects the input and output token pricing for the latest Mistral model and adds the newly released mistral-medium-2505 version.

## Title

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues

<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [ ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring
📖 Documentation
🚄 Infrastructure
✅ Test

## Changes

regismesquita committed:
Update mistral-medium prices and context sizes

While testing the Mistral model, I noticed a discrepancy in the pricing shown on the logs screen. After reviewing the code, I confirmed that the pricing values were incorrect.

This PR corrects the input and output token pricing for the latest Mistral model and adds the newly released mistral-medium-2505 version.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10729) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10729) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10729) it.</sub>

regismesquita committed:
Adds tool calling flag to mistral-medium

regismesquita committed:
Adds mistral-medium price updates to the main model price file

regismesquita committed:
Merge remote-tracking branch 'berriai/main' into patch-1

emerzon reviewed (commented):
The context sizes are incorrect - The previous values were correct.

regismesquita commented:
Mistral medium latest is only 32k?! Every place that I checked shows 128k.

regismesquita commented:
Does mistral/mistral-medium points to the old one?

regismesquita committed:
Update model_prices_and_context_window_backup.json

sets mistral medium alias to the old values as it probably points to the old version.

regismesquita committed:
Update model_prices_and_context_window.json

regismesquita commented:
@emerzon I have reverted mistral/mistral-medium to the old values, and now only the `-latest` alias and the `-2503` aliases are updated/created, I am keeping the 128k value, I have asked around and I am trying to confirm the actual values, if I can't find any I will do exploratory requests later today.

emerzon commented:
128k is for input tokens - not output tokens.

regismesquita commented:
@emerzon can you share how are you sure of that? I couldn't confirm that anywhere in their docs.

regismesquita committed:
Update model_prices_and_context_window_backup.json

regismesquita committed:
Update model_prices_and_context_window.json

regismesquita commented:
@emerzon haven't got any response yet, and I do think that it is less than 128k but since I haven't got any response from them I prefer to err on the safe side. so I changed it down to 8191 , and I can follow up later if I found any evidence against it.

emerzon commented:
LGTM

emerzon reviewed (approved):
LGTM

regismesquita commented:
I fixed the failing CI checks by adding missing module files and a test for the Mistral pricing. Here is the patch:

```patch
From abd50b9179d68780d5d5337a3fa664c6b9457d9d Mon Sep 17 00:00:00 2001
From: openhands <openhands@all-hands.dev>
Date: Sat, 17 May 2025 09:19:22 +0000
Subject: [PATCH] Add test for Mistral pricing and fix missing module

---
 litellm/llms/nvidia_nim/__init__.py      |  1 +
 litellm/llms/nvidia_nim/chat/__init__.py |  1 +
 tests/litellm/test_mistral_pricing.py    | 38 ++++++++++++++++++++++++
 3 files changed, 40 insertions(+)
 create mode 100644 litellm/llms/nvidia_nim/__init__.py
 create mode 100644 litellm/llms/nvidia_nim/chat/__init__.py
 create mode 100644 tests/litellm/test_mistral_pricing.py

diff --git a/litellm/llms/nvidia_nim/__init__.py b/litellm/llms/nvidia_nim/__init__.py
new file mode 100644
index 000000000..199cdd846
--- /dev/null
+++ b/litellm/llms/nvidia_nim/__init__.py
@@ -0,0 +1 @@
+# NVIDIA NIM Module
\ No newline at end of file
diff --git a/litellm/llms/nvidia_nim/chat/__init__.py b/litellm/llms/nvidia_nim/chat/__init__.py
new file mode 100644
index 000000000..83f8dbe15
--- /dev/null
+++ b/litellm/llms/nvidia_nim/chat/__init__.py
@@ -0,0 +1 @@
+# NVIDIA NIM Chat Module
\ No newline at end of file
diff --git a/tests/litellm/test_mistral_pricing.py b/tests/litellm/test_mistral_pricing.py
new file mode 100644
index 000000000..d75458abb
--- /dev/null
+++ b/tests/litellm/test_mistral_pricing.py
@@ -0,0 +1,38 @@
+import sys
+import os
+import json
+import pytest
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../..")))
+
+from litellm.utils import get_litellm_params
+
+def test_mistral_medium_latest_pricing():
+    """
+    Test that the pricing for mistral-medium-latest is correct
+    """
+    # Load the model prices
+    with open("litellm/model_prices_and_context_window_backup.json", "r") as f:
+        model_prices = json.load(f)
+    
+    # Check mistral-medium-latest pricing
+    model_info = model_prices.get("mistral/mistral-medium-latest", {})
+    
+    assert model_info.get("max_input_tokens") == 131072, "Incorrect max_input_tokens"
+    assert model_info.get("input_cost_per_token") == 0.0000004, "Incorrect input_cost_per_token"
+    assert model_info.get("output_cost_per_token") == 0.000002, "Incorrect output_cost_per_token"
+
+def test_mistral_medium_2505_pricing():
+    """
+    Test that the pricing for mistral-medium-2505 is correct
+    """
+    # Load the model prices
+    with open("litellm/model_prices_and_context_window_backup.json", "r") as f:
+        model_prices = json.load(f)
+    
+    # Check mistral-medium-2505 pricing
+    model_info = model_prices.get("mistral/mistral-medium-2505", {})
+    
+    assert model_info.get("max_input_tokens") == 131072, "Incorrect max_input_tokens"
+    assert model_info.get("input_cost_per_token") == 0.0000004, "Incorrect input_cost_per_token"
+    assert model_info.get("output_cost_per_token") == 0.000002, "Incorrect output_cost_per_token"
\ No newline at end of file
-- 
2.39.5


```

regismesquita committed:
Merge remote-tracking branch 'berriai/main' into patch-1

regismesquita committed:
Merge remote-tracking branch 'berriai/main' into patch-1

regismesquita commented:
synced with the upstream.

ishaan-jaff reviewed (approved):
Thanks for reviewing @emerzon
